src/GSMSSLClient.cpp

```diff
@@ -133,3 +133,43 @@ int GSMSSLClient::connect(const char* host, uint16_t port)
 
   return connectSSL(host, port);
 }
+
+void GSMSSLClient::setPrivateCertificate(const uint8_t* cert, const char* name, size_t size) {
+  MODEM.sendf("AT+USECMNG=0,1,\"%s\",%d", name, size);
+  MODEM.waitForResponse(1000);
+
+  MODEM.write(cert, size);
+  MODEM.waitForResponse(1000);
+}
+
+void GSMSSLClient::setPrivateKey(const uint8_t* key, const char*name, size_t size) {
+
+  MODEM.sendf("AT+USECMNG=0,2,\"%s\",%d", name, size);
+  MODEM.waitForResponse(1000);
+  MODEM.write(key, size);
+  MODEM.waitForResponse(1000);
+}
+
+void GSMSSLClient::setServerName(const char* name) {
+  MODEM.sendf("AT+USECPRF=0,3,\"%s\"", name);
+  MODEM.waitForResponse(100);
+}
+
+void GSMSSLClient::setClientName(const char* name) {
+  MODEM.sendf("AT+USECPRF=0,5,\"%s\"", name);
+  MODEM.waitForResponse(100);
+}
+
+void GSMSSLClient::setKeyName(const char* name) {
+  MODEM.sendf("AT+USECPRF=0,6,\"%s\"", name);
+  MODEM.waitForResponse(100);
+}
+
+void GSMSSLClient::setUserRoots(const GSMRootCert * userRoots, size_t size) {
+  for(int i=0; i<14; i++) {
```

Comment:
Is 14 the maximum number of supported certs? In that case it would be better to handle with a define; otherwise a sizeof would be better.

Reply:
yes are the snatadrd used by us, i can remove and replace with the same sie of the roots vector maybe make more sense, remove also the string arrey and put the the name attribute of the gsmrootcert srtucture

Reply:
i modified in order to be able to be reused when the array is changed with new one